payments/tests/test_payments_models.py

```diff
@@ -387,6 +389,62 @@ def test_order_berth_lease_right_price_for_full_season(harbor):
             assert product_price == order_price
 
 
+def test_order_winter_storage_lease_right_price_for_full_season(winter_storage_area):
+    services = {
+        "summer_storage_for_docking_equipment": True,
+        "summer_storage_for_trailers": random_bool(),
+    }
+    for service, create in services.items():
+        if create:
+            # Using PriceUnits.AMOUNT to simplify testing
+            AdditionalProductFactory(
+                service=ProductServiceType(service), price_unit=PriceUnits.AMOUNT
+            )
+
+    section = WinterStorageSectionFactory(**services, area=winter_storage_area)
+    lease = WinterStorageLeaseFactory(
+        place=WinterStoragePlaceFactory(winter_storage_section=section)
+    )
+    product = WinterStorageProductFactory(winter_storage_area=winter_storage_area)
+    order = Order.objects.create(product=product, customer=lease.customer, lease=lease)
+
+    for service, create in services.items():
+        if create:
+            additional_product = AdditionalProduct.objects.filter(
+                service=ProductServiceType(service),
+                price_unit=PriceUnits.AMOUNT,
+                period=PeriodType.SEASON,
+            ).first()
+            OrderLine.objects.create(order=order, product=additional_product)
+
+    assert order.lease.start_date == calculate_winter_storage_lease_start_date()
+    assert order.lease.end_date == calculate_winter_storage_lease_end_date()
```

Comment:
if you are not freezing time, you might get partial season leases, if you run the test between 15.9 and 9.6.

Reply:
same problem would happen not just in tests :derp: we should cover those cases as well in the `calculate_product_partial_season_price`

Reply:
> same problem would happen not just in tests :derp: we should cover those cases as well in the calculate_product_partial_season_price

What do you mean? 🤨

Reply:
if you run the `calculate_product_partial_season_price` on 8.6, you would get only 2 days in the season

Reply:
But wouldn't that be the case? I mean, if a lease were to be generated on 8.6, then the season end would be 10.6, unless a different date is specified..? Still a bit confused 😅

Reply:
yes, lease should be 2 days indeed in that case. but season should always start on 15.9 for the proper price calculation. with this approach any leas (even if it is a 2 day lease) will be charged full season price. in other words: we need something similar to [this](https://github.com/City-of-Helsinki/berth-reservations/blob/develop/leases/utils.py#L57)

Reply:
But wouldn't [this](https://github.com/City-of-Helsinki/berth-reservations/blob/develop/leases/utils.py#L77-L80) be enough then?
```python
def calculate_winter_storage_lease_start_date() -> date:
    """
    Return the date when the "winter" lease season should start
    The season should start by default on 15.9 (one day after the "summer" season ends)
    """

    # Return the earliest date between the default start date or today
    today = date.today()
    default = max(date(day=15, month=9, year=today.year), today)

    if today < date(day=10, month=6, year=today.year):
        return today

    return default
```

Reply:
Scratch that, I see the problem now. I'll do some local testing and work on the fix 🦖

Reply:
a good unit test for this would be
```
create a berth product with period type SEASON and cost of e.g. 1 eur per day (268 eur per winter season on non-leap years)
create a lease for 15 days
assert order.price == Decimal("15.00")
```